### PR TITLE
feat: translate Docs to PT-BR

### PR DESCRIPTION
**Pull request by developerfred**
#233 

## docs
* [x]  [`Nightly Builds.md`](https://github.com/microsoft/TypeScript-Website/blob/v2/packages//documentation/copy/en/Nightly%20Builds.md)
* [x]  [`Creating DTS files From JS.md`](https://github.com/microsoft/TypeScript-Website/blob/v2/packages//documentation/copy/en/javascript/Creating%20DTS%20files%20From%20JS.md)
* [x]  [`Intro to JS with TS.md`](https://github.com/microsoft/TypeScript-Website/blob/v2/packages//documentation/copy/en/javascript/Intro%20to%20JS%20with%20TS.md)
* [x]  [`React.md`](https://github.com/microsoft/TypeScript-Website/blob/v2/packages//documentation/copy/en/tutorials/React.md)
* [x]  [`TypeScript Tooling in 5 minutes.md`](https://github.com/microsoft/TypeScript-Website/blob/v2/packages//documentation/copy/en/tutorials/TypeScript%20Tooling%20in%205%20minutes.md)

### CODE REVIEW

**Comment by orta** [comment]
This is the first PR to add any handbook translations, so you might get me pushing changes in this PR adding CI infra stuff

**Comment by developerfred** [comment]
> This is the first PR to add any handbook translations, so you might get me pushing changes in this PR adding CI infra stuff

ok, but you can give me the list of files that I do the translation.

File: packages/documentation/copy/pt/Nightly Builds.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
+---
+title: Compilação Noturna
+layout: docs
+permalink: /docs/handbook/nightly-builds.html
+oneline: Como usar uma compilação noturna de TypeScript
+translatable: true
+---
+
+Uma compilação noturna do [`master` do TypeScript](https://github.com/Microsoft/TypeScript/tree/master) branch é publicado da meia-noite PST no npm.
+Veja como você pode obtê-lo e usá-lo com suas ferramentas.
```
Comment (danilofuchs) [inline packages/documentation/copy/pt/Nightly Builds.md]:
```suggestion
Uma compilação noturna da branch [`master` do TypeScript](https://github.com/Microsoft/TypeScript/tree/master) é publicada à meia-noite PST no npm.
Veja como você pode obtê-la e usá-la com suas ferramentas.
```

File: packages/documentation/copy/pt/Nightly Builds.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+"typescript_tsdk": "<caminho para sua pasta>/node_modules/typescript/lib"
+```
+
+Mais informações estão disponíveis em [Plug-in do TypeScript para documentação de instalação do Sublime Text](https://github.com/Microsoft/TypeScript-Sublime-Plugin#installation).
+
+### Visual Studio 2013 and 2015
+
+> Observação: a maioria das alterações não exige a instalação de uma nova versão do plug-in VS TypeScript.
+
+A construção noturna atualmente não inclui a configuração completa do plug-in, mas estamos trabalhando na publicação de um instalador todas as noites também.
```
Comment (danilofuchs) [inline packages/documentation/copy/pt/Nightly Builds.md]:
```suggestion
A compilação noturna atualmente não inclui a configuração completa do plug-in, mas estamos trabalhando na publicação de um instalador todas as noites também.
```

File: packages/documentation/copy/pt/Nightly Builds.md [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+1. Download the [VSDevMode.ps1](https://github.com/Microsoft/TypeScript/blob/master/scripts/VSDevMode.ps1) script.
```
Comment (danilofuchs) [inline packages/documentation/copy/pt/Nightly Builds.md]:
```suggestion
1. Baixe o script [VSDevMode.ps1](https://github.com/Microsoft/TypeScript/blob/master/scripts/VSDevMode.ps1).
```

File: packages/documentation/copy/pt/javascript/Intro to JS with TS.md [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+layout: docs
+permalink: /docs/handbook/intro-to-js-ts.html
+oneline: Como adicionar verificação de tipo a arquivos JavaScript usando TypeScript
+translatable: true
+---
+
+O sistema de tipos no TypeScript tem diferentes níveis de rigidez ao trabalhar com uma base de código:
+
+- Um sistema de tipo baseado apenas em inferência com código JavaScript
+- Digitação incremental em JavaScript [via JSDoc](/docs/handbook/jsdoc-supported-types.html)
```
Comment (danilofuchs) [inline packages/documentation/copy/pt/javascript/Intro to JS with TS.md]:
```suggestion
- Tipagem incremental em JavaScript [via JSDoc](/docs/handbook/jsdoc-supported-types.html)
```

File: packages/documentation/copy/pt/javascript/Intro to JS with TS.md [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+- Usando `// @ts-check` em um arquivo JavaScript
+- Código TypeScript
+- TypeScript com [`strict`](/tsconfig#strict) habilitado
+
+Cada etapa representa um movimento em direção a um sistema de tipo mais seguro, mas nem todo projeto precisa desse nível de verificação.
+
+## TypeScript com JavaScript
+
+Isso ocorre quando você usa um editor que usa TypeScript para fornecer ferramentas como autocompletar, pular para o símbolo e ferramentas de refatoração como renomear.
+A [home](/) possui uma lista de editores que possuem plugins TypeScript.
```
Comment (danilofuchs) [inline packages/documentation/copy/pt/javascript/Intro to JS with TS.md]:
```suggestion
A [página inicial](/) possui uma lista de editores que possuem plugins TypeScript.
```

File: packages/documentation/copy/pt/javascript/Intro to JS with TS.md [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+
+```js twoslash
+/** @type {number} */
+var x;
+
+x = 0; // OK
+x = false; // OK?!
+```
+
+Você pode encontrar a lista completa de padrões JSDoc suportados [em JSDoc Supported Types](/docs/handbook/jsdoc-supported-types.html).
```
Comment (danilofuchs) [inline packages/documentation/copy/pt/javascript/Intro to JS with TS.md]:
```suggestion
Você pode encontrar a lista completa de padrões JSDoc suportados [em tipos suportados de JSDoc](/docs/handbook/jsdoc-supported-types.html).
```

File: packages/documentation/copy/pt/javascript/Intro to JS with TS.md [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+// @ts-check
+/** @type {number} */
+var x;
+
+x = 0; // OK
+// @ts-expect-error
+x = false; // Not OK
+```
+
+Para saber mais sobre como o JavaScript é interpretado pelo TypeScript, leia [How TS Type Checks JS](/docs/handbook/type-checking-javascript-files.html)
```
Comment (danilofuchs) [inline packages/documentation/copy/pt/javascript/Intro to JS with TS.md]:
```suggestion
Para saber mais sobre como o JavaScript é interpretado pelo TypeScript, leia [Como TS Checa tipos de JS](/docs/handbook/type-checking-javascript-files.html)
```

File: packages/documentation/copy/pt/release-notes/Creating DTS files From JS.md [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+layout: docs
+permalink: /docs/handbook/declaration-files/dts-from-js.html
+oneline: "Como adicionar geração d.ts a projetos JavaScript"
+translatable: true
+---
+
+[Com o TypeScript 3.7](/docs/handbook/release-notes/typescript-3-7.html#--declaration-and---allowjs),
+TypeScript adicionou suporte para gerar arquivos .d.ts de JavaScript usando a sintaxe JSDoc.
+
+Essa configuração significa que você pode ter a experiência de editor de editores habilitados para TypeScript sem portar seu projeto para TypeScript ou ter que manter arquivos .d.ts em sua base de código.
```
Comment (danilofuchs) [inline packages/documentation/copy/pt/release-notes/Creating DTS files From JS.md]:
```suggestion
Essa configuração significa que você pode ter a experiência de editores habilitados para TypeScript sem portar seu projeto para TypeScript ou ter que manter arquivos .d.ts em sua base de código.
```

File: packages/documentation/copy/pt/release-notes/Creating DTS files From JS.md [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+Você pode aprender como fazer isso em nossa [página de instalação](/download).
+
+### TSConfig
+
+O TSConfig é um arquivo json5 que configura ambos os sinalizadores do compilador e declara onde encontrar os arquivos.
+Neste caso, você vai querer um arquivo como o seguinte:
+
+```json5
+{
+  // Change this to match your project
```
Comment (danilofuchs) [inline packages/documentation/copy/pt/release-notes/Creating DTS files From JS.md]:
```suggestion
  // Mude isso para corresponder ao seu projeto
```

File: packages/documentation/copy/pt/release-notes/Creating DTS files From JS.md [hunk 10]
```diff
@@ -0,0 +1,88 @@
...
+  include: ["src/**/*"],
+
+  compilerOptions: {
+    // Tells TypeScript to read JS files, as
+    // normally they are ignored as source files
```
Comment (danilofuchs) [inline packages/documentation/copy/pt/release-notes/Creating DTS files From JS.md]:
```suggestion
    // Diz para o TypeScript ler arquivos JS.
    // Normalmente, seriam ignorados como arquivos fonte
```

File: packages/documentation/copy/pt/release-notes/Creating DTS files From JS.md [hunk 11]
```diff
@@ -0,0 +1,88 @@
...
+    allowJs: true,
+    // Generate d.ts files
```
Comment (danilofuchs) [inline packages/documentation/copy/pt/release-notes/Creating DTS files From JS.md]:
```suggestion
    // Gerar arquivos d.ts
```

File: packages/documentation/copy/pt/release-notes/Creating DTS files From JS.md [hunk 12]
```diff
@@ -0,0 +1,88 @@
...
+    declaration: true,
+    // This compiler run should
+    // only output d.ts files
```
Comment (danilofuchs) [inline packages/documentation/copy/pt/release-notes/Creating DTS files From JS.md]:
```suggestion
    // A compilação só gerará arquivos
    // d.ts na saída
```

File: packages/documentation/copy/pt/release-notes/Creating DTS files From JS.md [hunk 13]
```diff
@@ -0,0 +1,88 @@
...
+    emitDeclarationOnly: true,
+    // Types should go into this directory.
+    // Removing this would place the .d.ts files
+    // next to the .js files
```
Comment (danilofuchs) [inline packages/documentation/copy/pt/release-notes/Creating DTS files From JS.md]:
```suggestion
    // Tipos devem ir neste diretório.
    // Remover isso colocará arquivos .d.ts
    // ao lado dos arquivos .js
```

File: packages/documentation/copy/pt/release-notes/Creating DTS files From JS.md [hunk 14]
```diff
@@ -0,0 +1,88 @@
...
+```
+
+## Execute o compilador
+
+Você pode aprender como fazer isso em nossa [página de instalação](/download).
+Você quer ter certeza de que esses arquivos estão incluídos em seu pacote se você tiver os arquivos `.gitignore` em seu projeto.
+
+## Editando o package.json
+
+TypeScript replica a resolução do nó para módulos em um `package.json`, com uma etapa adicional para localizar arquivos `.d.ts`.
```
Comment (danilofuchs) [inline packages/documentation/copy/pt/release-notes/Creating DTS files From JS.md]:
```suggestion
TypeScript replica a resolução do Node para módulos em um `package.json`, com uma etapa adicional para localizar arquivos `.d.ts`.
```

File: packages/documentation/copy/pt/release-notes/Creating DTS files From JS.md [hunk 15]
```diff
@@ -0,0 +1,88 @@
...
+A grosso modo, a resolução verificará primeiro o campo opcional `"types"`, depois o campo `"main"` e, finalmente, tentará `index.d.ts` na raiz.
+
+| Package.json              | Localização do padrão .d.ts      |
+| :------------------------ | :----------------------------- |
+| No "types" field          | checks "main", then index.d.ts |
+| "types": "main.d.ts"      | main.d.ts                      |
+| "types": "./dist/main.js" | ./main/main.d.ts               |
```
Comment (danilofuchs) [inline packages/documentation/copy/pt/release-notes/Creating DTS files From JS.md]:
```suggestion
| Package.json              | Localização padrão do .d.ts      |
| :------------------------ | :----------------------------- |
| Sem campo "types"         | Checa "main", então index.d.ts |
| "types": "main.d.ts"      | main.d.ts                      |
| "types": "./dist/main.js" | ./main/main.d.ts               |
```

File: packages/documentation/copy/pt/release-notes/Creating DTS files From JS.md [hunk 16]
```diff
@@ -0,0 +1,88 @@
...
+
+If absent, then "main" is used
```
Comment (danilofuchs) [inline packages/documentation/copy/pt/release-notes/Creating DTS files From JS.md]:
```suggestion
Se estiver ausente, "main" é usado
```

File: packages/documentation/copy/pt/release-notes/Creating DTS files From JS.md [hunk 17]
```diff
@@ -0,0 +1,88 @@
...
+
+| Package.json             | Localização do padrão .d.ts |
+| :----------------------- | :------------------------ |
+| No "main" field          | index.d.ts                |
+| "main":"index.js"        | index.d.ts                |
+| "main":"./dist/index.js" | ./dist/index.d.ts         |
```
Comment (danilofuchs) [inline packages/documentation/copy/pt/release-notes/Creating DTS files From JS.md]:
```suggestion
| Package.json             | Localização padrão do .d.ts |
| :----------------------- | :------------------------ |
| Sem campo "main"         | index.d.ts                |
| "main":"index.js"        | index.d.ts                |
| "main":"./dist/index.js" | ./dist/index.d.ts         |
```

File: packages/documentation/copy/pt/tutorials/TypeScript Tooling in 5 minutes.md [hunk 18]
```diff
@@ -0,0 +1,186 @@
...
+let user = "Jane User";
+
+document.body.textContent = greeter(user);
+```
+
+## Digite anotações
+
+As anotações de tipo no TypeScript são maneiras leves de registrar o contrato pretendido da função ou variável.
+Neste caso, pretendemos que a função saudador seja chamada com um único parâmetro de string.
+Podemos tentar alterar o saudador de chamada para passar uma matriz em vez disso:
```
Comment (danilofuchs) [inline packages/documentation/copy/pt/tutorials/TypeScript Tooling in 5 minutes.md]:
```suggestion
As anotações de tipo no TypeScript são maneiras leves de registrar o contrato pretendido da função ou variável.
Neste caso, pretendemos que a função greeter seja chamada com um único parâmetro de string.
Podemos tentar alterar a chamada de greeter para passar uma matriz em vez disso:
```

File: packages/documentation/copy/pt/tutorials/TypeScript Tooling in 5 minutes.md [hunk 19]
```diff
@@ -0,0 +1,186 @@
...
+document.body.textContent = greeter(user);
+```
+
+Recompilando, você verá um erro:
+
+```shell
+error TS2345: Argument of type 'number[]' is not assignable to parameter of type 'string'.
+```
+
+Da mesma forma, tente remover todos os argumentos para a chamada de saudação.
```
Comment (danilofuchs) [inline packages/documentation/copy/pt/tutorials/TypeScript Tooling in 5 minutes.md]:
```suggestion
Da mesma forma, tente remover todos os argumentos para a chamada de greeter.
```

File: packages/documentation/copy/pt/tutorials/TypeScript Tooling in 5 minutes.md [hunk 20]
```diff
@@ -0,0 +1,186 @@
...
+</html>
+```
+
+Abra `greeter.html` no navegador para executar seu primeiro aplicativo da Web TypeScript simples!
+
+Opcional: Abra `greeter.ts` no Visual Studio ou copie o código para o playground do TypeScript.
+Você pode passar o mouse sobre os identificadores para ver seus tipos.
+Observe que, em alguns casos, esses tipos são inferidos automaticamente para você.
+Digite novamente a última linha e veja as listas de completamento e a ajuda de parâmetros com base nos tipos de elementos DOM.
+Coloque o cursor sobre a referência da função saudação e pressione F12 para ir para sua definição.
```
Comment (danilofuchs) [inline packages/documentation/copy/pt/tutorials/TypeScript Tooling in 5 minutes.md]:
```suggestion
Coloque o cursor sobre a referência da função greeter e pressione F12 para ir para sua definição.
```

**Review by danilofuchs — CHANGES_REQUESTED** [review]
Great job!
I have some suggestions

File: packages/documentation/copy/pt/javascript/Intro to JS with TS.md (hunk 6 above)
Comment (danilofuchs) [reply packages/documentation/copy/pt/javascript/Intro to JS with TS.md]: Please translate this phrase

File: packages/documentation/copy/pt/javascript/Intro to JS with TS.md (hunk 7 above)
Comment (danilofuchs) [reply packages/documentation/copy/pt/javascript/Intro to JS with TS.md]: Here as well. These are shown as-is, with hyperlinks, so the user must have the information in Portuguese

**Comment by danilofuchs** [comment]
@developerfred I believe you forgot to update some parts. 

Expand the 'hidden conversations' part to reveal the other suggestions:
image

**Comment by developerfred** [comment]
> @developerfred I believe you forgot to update some parts.
> 
> Expand the 'hidden conversations' part to reveal the other suggestions:
> image

Thanks! ✅

**Comment by danilofuchs** [comment]
LGTM

**Comment by orta** [comment]
I've rebased this, lets hope it fixes CI

**Comment by khaosdoctor** [comment]
CI is Fixed!

**Review by khaosdoctor — APPROVED** [review]
LGTM

**Comment by khaosdoctor** [comment]
LGTM

**Comment by khaosdoctor** [comment]
@orta Either the bot is not working, or the file is not in my scope

**Comment by orta** [comment]
Yeah, I'm bad with rebases it seems

**Pull request merged**